en/05_Contributing/05_Coding_Conventions/01_PHP_Coding_Conventions.md

```diff
@@ -6,190 +6,157 @@ iconBrand: php
+- For third-party classes, [use `Injector` directly](/developer_guides/extending/injector/#basic-usage) to instantiate objects rather than the `new` keyword.
+- For unit tests, always use the `new` keyword when instantiating objects unless you are explicitly testing the injector logic.
 
-It is preferable to use parameterised queries whenever necessary to provide conditions
-to a SQL query, where values placeholders are each replaced with a single unquoted question mark.
-If it's absolutely necessary to use literal values in a query make sure that values
-are single quoted.
+## Extensions and traits
 
+- Use the [`Extension`](api:SilverStripe\Core\Extension) class for extending classes, including `DataObject` subclasses.
+- Do not use the `DataExtension` class, it will be deprecated in a future release.
 
-```php
-MyClass::get()->where("\"Title\" = 'my title'");
-```
-
-Use [ANSI SQL](https://en.wikipedia.org/wiki/SQL#Standardization) format where possible.
+Use a trait instead of an [`Extension`](api:SilverStripe\Core\Extension) when the composable functionality:
 
-## Secure Development
+- Relies on per-instance properties, because they cannot be used with extensions
+- Does not add or modify configuration properties or values
+- Is not expected to be applied to third-party code.
 
-See [security](/developer_guides/security) for conventions related to handing security permissions.
+Use an [`Extension`](api:SilverStripe\Core\Extension) instead of a trait when the composable functionality:
 
-## Related
+- Does not rely on per-instance properties
+- Adds or modifies configuration properties or values
+- Could be expected to be applied to third-party code.
 
- * [PHPCS Coding Standard](https://github.com/silverstripe/silverstripe-framework/blob/4/phpcs.xml.dist)
- * [JavaScript Coding Conventions](/contributing/javascript_coding_conventions)
- * [Reference: CMS Architecture](/developer_guides/customising_the_admin_interface/cms_architecture)
- 
+### Extension hooks
+
+- For hooks that update the value that's returned, name the hook `update{MethodName}` where `MethodName` has any prefix `get` removed. For example `getMyValue()` would have an extension hook named `updateMyValue()`.
+- For hook that update a variable in the middle of a method, name the hook `update{Variable}` where `Variable` is the name of the variable being updated. For example if a method has a variable called `$myValue` then the hook would be named `updateMyValue()`.
+- For hooks that update method parameters before running the body of the method, name the hook `onBefore{MethodName}`. For instance if a method calls `processSomething()` call the hook `onBeforeProcessSomething()`.
+- If the method has an `onBefore` hook and also has a hook to update value that's returned, then second hook should be called `onAfter{MethodName}` rather than `update{MethodName}`.
```

Comment:
This onafter one is a bit confusing. I'd prefer it just be the same wording as for the `onBefore` one.

Reply:
Updated